src/main/scala/io/epiphanous/flinkrunner/util/aws/AWSSigner.scala

```diff
@@ -72,6 +72,17 @@ class AWSSigner(
               .withMultiValueQueryParams(request.multiParams)
           )
         )
+      }else if(awsEndpoint.isDefined) {
```

Comment:
I don't understand this if/else.  Also, given you're using an Optionn for aws variable, you should not be checking nulls.  That's what an option is, a safer way to handle nulls.

Reply:
It was a little messy.. Fixed it.
We need to check if awsEndpoint is empty or defined so that we can call appropriate code to form the final endpoint for request

Reply:
I don't see a fix? did you push?

Reply:
I did make the necessary changes as per me.. As we have different ways of forming endpoint in different case so I need to check if it is awsEndpoint is defined or not.

Reply:
its still checking for the Option to be null though

Reply:
Yeah. because based on the value awsEndpoint we need to decide which code to run.. I have given it as an Option because localendpoint will be null in case we are running in sandbox or any other env other than locally on docker.